### ISSUE TITLE
Add tool to catch the shadowing of _ (gettext)

### ISSUE THREAD

**Issue by bmispelon** [django/djangoproject.com#1721]
With the fantastic work that's been going on to make the site translatable, we've been using the `_` alias for gettext more and more throughout the codebase.

Unfortunately, the name `_` is also sometimes used to create throwaway variables, and this can lead to crashes (see e9952130db15e603477a5bc2a3a8f202c99cbd65).

It would be nice if he had a tool to warn us when reusing the name `_` for a throwaway variable in a file that already has `_` as an alias for `gettext`. I'm hoping there's already a `flake8` plugin for it, or something similar.

**Comment by bmispelon** [comment]
@marksweb tagging you in just in case you know of existing solutions. This is something I didn't think of when doing the code review, I'll try to pay more attention next time.

**Comment by marksweb** [comment]
Oh sorry I missed this one.

Historically when I've had this issue I change the import alias because the underscore is the unused variable standard.

**Comment by bmispelon** [comment]
For now I've used `git grep -E '\b_[^(_a-zA-Z9-9]' -- '*.py'` to check for usage of the throwaway `_` variable and I'm confident that we don't have other instances of this error currently.

I've looked around a bit and haven't found any available plugins, so this might need to be custom-made 🤔

**Comment by marksweb** [comment]
@bmispelon yeah I've not found anything either. Could likely set that up as a workflow action to run on PRs

**Comment by ulgens** [comment]
I think we are looking for https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/redefined-outer-name.html ruff has multiple "shadowing" or "redefining" checks but I'm not sure which one would cover the case here.

**Comment by aksharabhardwaj766-commits** [comment]
Hi! I’d be happy to look into this.
I’ll first check whether Ruff or Pylint already detects this specific case (shadowing _ when it’s imported as a gettext alias).
If neither tool covers it properly, I can prototype a small custom check and integrate it into CI.
Does that approach sound reasonable?

**Comment by mgax** [comment]
Ruff has [import-shadowed-by-loop-var (F402)](https://docs.astral.sh/ruff/rules/import-shadowed-by-loop-var/) which catches this:

```py
for _ in range(13):
    pass
```

But not this:

```py
_ = int(13)
```

There's a PR open to implement "Redefined outer names (PLW0621)", https://github.com/astral-sh/ruff/pull/15903, which has been dormant for more than a year now; I could not find anything else.
